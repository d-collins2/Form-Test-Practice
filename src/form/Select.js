import React from 'react';
import { Form } from 'react-bootstrap'


const Select = ({value, handleChange, formError, errorClass, name, controlId, serviceTypes}) => {
    return (
      <Form.Group
          className={errorClass(formError)}
          controlId="controlId">
          <Form.Control
              required
              as="select"
              onChange={handleChange}
              name= "serviceType"
              value={value}>
              <option value="">Select Service Type</option>
              {serviceTypes && serviceTypes.map((type, id) => {
                  return <option key={id} value={type.display_name}>{type.display_name}</option>
              })}
          </Form.Control>
            <span className="help-block">{formError}</span>
            <div className="text-right">
                <Form.Text className="text-red">Required</Form.Text>
            </div>
      </Form.Group>

    )
}

Select.defaultProps = {
    className: 'form-group',
    labelFor: 'checkbox',
    value: [],
    handleChange: function(){},
    label: 'Select'
}

export default Select;
