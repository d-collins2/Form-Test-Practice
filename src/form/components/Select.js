import React from "react";
import { Form } from 'react-bootstrap'
import Options from "./Options.js";

const Select = ({ controlId, formError, handleChange, name, serviceTypes, value }) => {
    function errorClass(error) {
        if(error){
            return( error.length === 0 ? '' : 'has-error');
        }
    }
  	return (
    		<Form.Group className={errorClass(formError)} id="controlId">
      			<Form.Control
        				required
                as="select"
                name={name}
        				onChange={handleChange}
        				value={value}>
        			  <option value="">Select Service Type</option>
        				<Options options={serviceTypes}/>
      			</Form.Control>
      			<Form.Text className="help-block text-red">{formError}</Form.Text>
            <div className="text-right">
                <Form.Text>className="text-muted">Required</Form.Text>
            </div>
    		</Form.Group>
  	);
};

export default Select;
