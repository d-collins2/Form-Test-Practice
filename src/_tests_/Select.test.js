import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-bootstrap'
import { shallow } from 'enzyme';
import Select from '../form/components/Select';

describe('<Select />', function(){
    it('Should render correctly', function() {
        const component = renderer.create(<Select />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Should have one <select>', function() {
        const component = shallow(<Select />);
        expect(component.find(Form.Control).length).toBe(1);
    })

    it('Should be able to select one <option> from drop down', function() {
        const component = shallow(<Select />);
        expect(component.find('option').first().props().overlay);
    })
})
