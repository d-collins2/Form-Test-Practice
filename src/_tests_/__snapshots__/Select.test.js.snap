// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Should render correctly 1`] = `
<div
  className="form-group"
  id="controlId"
>
  <select
    className="form-control"
    required={true}
  >
    <option
      value=""
    >
      Select Service Type
    </option>
  </select>
  <small
    className="help-block text-red form-text"
  />
  <div
    className="text-right"
  >
    <small
      className="form-text"
    >
      className="text-muted"&gt;Required
    </small>
  </div>
</div>
`;
